hoverkite-protocol/src/response.rs

```diff
@@ -31,14 +30,41 @@ pub enum Report {
     ChargeState {
         charger_connected: bool,
     },
+    PowerOff,
 }
 
-impl Report {
+impl Response {
+    pub fn log_from_fmt(args: core::fmt::Arguments<'_>) -> Self {
+        let mut writer = TruncatingWriter(ArrayString::new());
+
+        match core::fmt::Write::write_fmt(&mut writer, args) {
```

Comment:
Can't you just `use core::fmt::Write` and then `writer.write_fmt(args)`?

Reply:
I think I was being careful about bringing loads of Write traits into scope, because there are 4 in use in this project and it gets confusing. In this case, this file isn't using any of the others, so it's probably fine. Changed.